src/OpenTelemetry.AutoInstrumentation/Instrumentation.cs

```diff
@@ -118,6 +119,15 @@ private static void OnExit(object sender, EventArgs e)
         Log("OpenTelemetry tracer exit.");
     }
 
+    private static void OnUnhandledException(object sender, UnhandledExceptionEventArgs args)
+    {
+        Log("UnhandledException event raised.");
```

Comment:
I assume that mostly issue will be not related to our code, but it could be useful to have a reason of the event here. Consider to log the data from `ExceptionObject`.

Reply:
I would also consider moving this logging inside the `if (args.IsTerminating)` block.

Reference: https://docs.microsoft.com/en-us/dotnet/standard/threading/exceptions-in-managed-threads

Reply:
Write now this log message is really just writing text to stdout. I don't know if it is something that we want to keep. We already have #334 to change the way we handle this logging, but I want to defer that discussion until we work on #334.

As of right now, I think there is value in logging a message outside of the `IsTerminating` check, because we have additional logging that occurs when `OnExit` is called inside of that check (but in the `OnExit` method). This could be particularly useful if we are able to capture the exception and add it to telemetry even if the application is not being terminated. I am purposely being vague about how we can report that exception, because we may not have a span available to report the exception on, so we may need to report the exception via the logging pipeline (when available). @pellared is your concern about logging outside of the `IsTerminating` check related to those messages possibly being noisy or a distraction, or not actionable by someone?

I considered adding the exception details to the current log message, but I wanted to delay that decision. Some people can be sensitive to where exception details get recorded, so I did not want to add the exception details to the log message without a way for someone to control those exception details. I'm open to other thoughts on this

Reply:
> @pellared is your concern about logging outside of the IsTerminating check related to those messages possibly being noisy or a distraction, or not actionable by someone?

Yes. Not a blocker.

> Some people can be sensitive to where exception details get recorded

Totally agree. I think it would violate OWASP logging guidelines.

Reply:
> > @pellared is your concern about logging outside of the IsTerminating check related to those messages possibly being noisy or a distraction, or not actionable by someone?
> 
> Yes. Not a blocker.

I think your concern is valid and it's safer for me to move the logging. If we ever do something outside of that check, it'll be easy for us to add logging later. It's easier for us to add something than to take away something later.